docroot/modules/custom/prisoner_hub_entity_access/src/EventSubscriber/QueryAccessSubscriber.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Drupal\prisoner_hub_entity_access\EventSubscriber;
+
+use Drupal\Core\Entity\EntityFieldManagerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\entity\QueryAccess\ConditionGroup;
+use Drupal\entity\QueryAccess\QueryAccessEvent;
+use Drupal\taxonomy\TermInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Class QueryAccessSubscriber.
+ *
+ * Uses the entity query access API (from the entity module).  To show/hide
+ * content based on prison and prison category fields.
+ */
+class QueryAccessSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The entity field manager.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface
+  */
+  protected $entityFieldManager;
+
+  /**
+   * The route match service.
+   *
+   * @var RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * The prison category field name.
+   *
+   * @var String
+   */
+  protected $prisonCategoryFieldName;
+
+  /**
+   * The prison category field name.
+   *
+   * @var String
+   */
+  protected $prisonFieldName;
+
+  public function __construct(EntityFieldManagerInterface $entity_field_manager, RouteMatchInterface $route_match, string $prison_field_name, string $prison_category_field_name) {
+    $this->entityFieldManager = $entity_field_manager;
+    $this->routeMatch = $route_match;
+    $this->prisonFieldName = $prison_field_name;
+    $this->prisonCategoryFieldName = $prison_category_field_name;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    // TODO: Do we want to run this for Taxonomy Terms?  Or other entity types?
```

Comment:
Is this `TODO` still outstanding?

Reply:
Yeah it is still outstanding.  Not sure if we want to apply the rule to taxonomy terms as well?  Currently we don't have the prison and prison category field on any taxonomy terms (other than the prison terms itself), so applying the rule wouldn't actually do anything yet.  But if we did, and then added those fields, it would start to work in the same way as it does for content.

Reply:
:+1: assuming it doesn't simplify this code, we should probably add that if/when we update the taxonomy terms 🙌